sycl/CMakeLists.txt

```diff
@@ -34,12 +34,20 @@ set(CLANG_VERSION "${CLANG_VERSION_MAJOR}.${CLANG_VERSION_MINOR}.${CLANG_VERSION
 set(LLVM_INST_INC_DIRECTORY "lib${LLVM_LIBDIR_SUFFIX}/clang/${CLANG_VERSION}/include")
 set(dst_dir ${LLVM_LIBRARY_OUTPUT_INTDIR}/clang/${CLANG_VERSION}/include)
 
-find_package(OpenCL)
+# Find OpenCL headers and libraries installed in the system and use them to
+# build SYCL runtime.
+# WARNING: use with caution, building SYCL runtime with OpenCL implementation
+# instead of Khronos ICD loader might cause build and/or portability issues.
+option(BUILD_WITH_SYSTEM_OPENCL OFF)
+
+if( (OpenCL_INCLUDE_DIR AND OpenCL_LIBRARY) OR BUILD_WITH_SYSTEM_OPENCL)
```

Comment:
Probably setting both `BUILD_WITH_SYSTEM_OPENCL` and `OpenCL_*` variables at the same time should give at least a warning to let the user know which option has higher priority.

Reply:
Interesting idea, but are you sure that warning is really needed?
I mean if some users really reads those messages from configure step, then that user will see what version was chosen. find_package(OpenCL) explicitly writes that. For example, on a system with OpenCL available in the system and cmake flags: 

-DOpenCL_INCLUDE_DIR=C:/Users/vklochko/ws/gs_public/OpenCL-Headers -DOpenCL_LIBRARY=C:/Users/vklochko/ws/gs_public/OpenCL-ICD-Loader/build/OpenCL.lib -DOpenCL_BUILD_WITH_SYSTEM_SDK=ON

Then cmake writes:
-- Found OpenCL: C:/Users/vklochko/ws/gs_public/OpenCL-ICD-Loader/build/OpenCL.lib (found version "2.2")

This message makes it pretty obvious that those 2 vars def have higher priority than system sdk.
Do you agree?